src/Dev/Tasks/LegacyThumbnailMigrationHelper.php

```diff
@@ -145,9 +152,13 @@ protected function migrateFolder(FlysystemAssetStore $store, Folder $folder)
 
             $oldResampledPath = $fileInfo['path'];
 
-            // Get "variant folders" inside _resampled folder.
-            $variantFolders = preg_replace('#.*_resampled/(.*)#', '$1', $fileInfo['dirname']);
-            $encodedVariants = explode(DIRECTORY_SEPARATOR, $variantFolders);
+            $parsedFileID = $legacyFileIDParser->parseFileID($oldResampledPath);
+
+            if (!$parsedFileID) {
+                throw new \LogicException(
```

Comment:
I thought about replacing this exception with an error log, so the entire job won't crash if it comes across an error. But that would have mean tweaking the directory deletion below.

Reply:
If people hit this on a large site, you could potentially be adding _days_ to a migration project, as you'd need to run, fix, run, fix a script that takes hours and hours. How likely is it that we'd hit this exception?

Reply:
Any nonsensical file inside a `_resampled` folder would trip this Exception, so I would expect mildly common for big projects that have been going a long time.

However this is not the primary migration tasks. It's meant to clean up any orphan variants after the main migration task is done. In practice, it's an optional task.

Switching to a logger would be a simple process. I just left the note for Ingo to ponder during his peer reviewing.

Reply:
I've flagged a ticket about the wider issue of "defensive coding" during file migration with Bryn, it's part of Exp Debt now: https://github.com/silverstripe/silverstripe-assets/issues/285. So basically: Don't abort task unless there's no other option